Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -73,48 +76,176 @@ public void RoslynConfig_NoAssemblies()
+            
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.pluginKey</Key>
+                  <Value>securitycsharpfrontend</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.pluginVersion</Key>
+                  <Value>1.42.0</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.staticResourceName</Key>
+                  <Value>SecurityAnalyzer.zip</Value>
+                </Setting>
+            """ : string.Empty;
+        var expectedSonarLintXml = $"""
+            <?xml version="1.0" encoding="UTF-8"?>
+            <AnalysisInput>
+              <Settings>
+                <Setting>
+                  <Key>sonar.{language}.analyzer.dotnet.pluginKey</Key>
+                  <Value>{language}</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.{language}.analyzer.dotnet.pluginVersion</Key>
+                  <Value>1.42.0</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.{language}.analyzer.dotnet.staticResourceName</Key>
+                  <Value>SonarAnalyzer.zip</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.{language}.testPropertyPattern</Key>
+                  <Value>foo</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.{language}.foo</Key>
+                  <Value>bar</Value>
+                </Setting>{securityProperties}
+              </Settings>
+              <Rules>
+                <Rule>
+                  <Key>{language}-S1116</Key>
+                  <Parameters>
+                    <Parameter>
+                      <Key>key</Key>
+                      <Value>value</Value>
+                    </Parameter>
+                  </Parameters>
+                </Rule>
+                <Rule>
+                  <Key>{language}-S1125</Key>
+                </Rule>
+              </Rules>
+              <Files>
+              </Files>
+            </AnalysisInput>
+            
+            """;
+        context.AssertCorrectAnalyzerSettings();
+        context.AssertNoWarningsOrErrors();
+        context.AssertInfoMessages($"Provisioning analyzer assemblies for {language}...");
+        context.AssertCorrectRulesets();
+        context.AssertExpectedAssemblies(@"c:\assembly1.dll", @"d:\foo\assembly2.dll");
+        var expectedPlugins = new List<Plugin>()
+        {
+            new() { Key = "wintellect", Version = "1.13.0", StaticResourceName = "wintellect.zip" },
+            new() { Key = language, Version = "1.42.0", StaticResourceName = "SonarAnalyzer.zip" },
+        };
+        if (language == RoslynAnalyzerProvider.CSharpLanguage)
+        {
+            expectedPlugins.Add(new() { Key = "securitycsharpfrontend", Version = "1.42.0", StaticResourceName = "SecurityAnalyzer.zip" });
+        }
```

Comment:
Calculation of `expectedPlugins` should be also extracted, as it will be used 3x